### PR TITLE
change license to GPL v3

### PR DESCRIPTION
**Pull request by erciccione**
The Apache 2 license [seems to be not compatible](https://www.apache.org/licenses/GPL-compatibility.html) with the AGPL v3 license of our core repo, as we initially thought. Even only linking to the AGPL code is considered a derivative work and bundling the two together would be also probably considered a derivative work.

We need to change the license of the GUI to something compatible, the only choice we have is AGPL v3 or GPL v3. We decided to go for the GPL, as it's slightly less restrictive.

We ask all past contributors to approve the change, so we can go ahead and merge this.

@premek @Tcobli @Randall-Coding @l0nelyc0w @xyzmaker123 @duriancrepe please leave a comment that says `ACK` or use github's approval of the PR. Thank you all :)

### CODE REVIEW

**Review by premek — APPROVED** [review]

**Comment by premek** [comment]
ACK

But my understanding is you can include Apache licensed component in an AGPL application as long as it stays AGPL. In this case I think Haveno could be AGPL and use Apache components. It probably already uses some java dependencies that are under MIT or Apache i think.

**Comment by erciccione** [comment]
@premek 

> But my understanding is you can include Apache licensed component in an AGPL application as long as it stays AGPL

That was my understanding too, but Apache is clear in this regard:

> This licensing incompatibility applies only when some Apache project software becomes a derivative work of some GPLv3 software, because then the Apache software would have to be distributed under GPLv3. This would be incompatible with ASF's requirement that all Apache software must be distributed under the Apache License 2.0.

The AGPL software cannot be relicensed to Apache 2, that's where the incompatibility arises.

**Comment by premek** [comment]
definitely not an expert either :)

> The Apache License is permissive; unlike copyleft licenses, **it does not require a derivative work** of the software, or modifications to the original, **to be distributed using the same license**. It still requires application of the same license to all unmodified parts. In every licensed file, original copyright, patent, trademark, and attribution notices must be preserved (excluding notices that do not pertain to any part of the derivative works). In every licensed file changed, a notification must be added stating that changes have been made to that file.

https://en.wikipedia.org/wiki/Apache_License

>> This licensing incompatibility applies only when some Apache project software becomes a derivative work of some GPLv3 software, because **then the Apache software would have to be distributed under GPLv3.** This would be incompatible with ASF's requirement that all Apache software must be distributed under the Apache License 2.0.

> The Apache 2 software cannot be relicensed to (A)GPL v3, that's where the incompatibility arises.

What this means IMO is that the derivative (the whole Haveno bundle in our case) would have to be AGPL and not Apache. They say it's not compatible with Apache software foundation requirement but not the license itself I think. Apache wants all their software to be licensed under Apache license but that shouldn't matter for Haveno as Haveno is not part of ASF, just using the license.

In short, change it, it doesn't matter, having just one license will be easier :)

**Comment by erciccione** [comment]
> What this means IMO is that the derivative (the whole Haveno bundle in our case) would have to be AGPL and not Apache. They say it's not compatible with Apache software foundation requirement but not the license itself I think

If that's the case then it would be in line with our initial thought (that apache 2 is compatible), but they talk about "Apache" software, which i interpreted as "Apache licensed" software, not software produced by the apache foundation. If that interpretation is not correct, then there should be no issue.

> In short, change it, it doesn't matter, having just one license will be easier :)

We would prefer to keep Apache 2 for the GUI, as it makes easier for people to contribute to it. We were thinking of changing only because of the possible incompatibility. If that's not a problem, there is no reason to change license.

**Comment by erciccione** [comment]
Thinking about it, one of the perks of the main features of the Apache 2 license is that larger works can be redistributed under a different license, so i was definitely wrong in thinking that derivative works need the same license. That's actually one of the main points of the Apache 2 license in the first place.

I got confused by the wording on the apache page I linked above. I don't think it's necessary to change license. Sorry for the misunderstanding. I'll close this PR and remove the pings to people. In the meantime i will investigate our licensing situation and if necessary i will reopen this PR. Thanks @premek for helping clarify :)

**Comment by Randall-Coding** [comment]
Whew glad I don't have to give an opinion on this :)  Unfortunately I have no understanding of legal / licensing issues.
